hpy/devel/src/runtime/argparse.c

```diff
@@ -39,45 +39,82 @@ int _HPyArg_ParseItem(HPyContext ctx, HPy current_arg, const char **fmt, va_list
         *output = current_arg;
         break;
     }
+    case 'N': {
+        HPy *output = va_arg(*vl, HPy *);
+        _BREAK_IF_OPTIONAL(current_arg);
+        *output = HPy_Dup(ctx, current_arg);
+        if (HPy_IsNull(*output))
+            return 0;
+        break;
+    }
     default:
         HPyErr_SetString(ctx, ctx->h_ValueError, "XXX: Unknown arg format code");
         return 0;
     }
     return 1;
 }
 
+static void
+skip_next_format_character(HPyContext ctx, const char **fmt, va_list *vl)
+{
+    if (!parse_item(ctx, HPy_NULL, fmt, vl))
+        HPy_FatalError(ctx, "argparse.c: skip_next_format_character failed");
+}
+
+static void
+clear_already_allocated_handles(HPyContext ctx, const char *fmt,
```

Comment:
it would be nice to have a test which check that this has been called, although it's probably hard to write. Maybe once we have a proper debug mode, it will be easier?